### PR TITLE
[v5] Add `spawn()` action

### PR DESCRIPTION
**Pull request by davidkpiano**

### CODE REVIEW

File: packages/core/src/actions/spawn.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+/**
+ *
+ * @param expr The expression function to evaluate which will be logged.
+ *  Takes in 2 arguments:
+ *  - `ctx` - the current state context
+ *  - `event` - the event that caused this action to be executed.
+ * @param label The label to give to the logged expression.
+ */
+export function spawn<
```
Comment (Andarist) [inline packages/core/src/actions/spawn.ts:67]: q: isn't this - in a way - just an exposed `invoke` action? :p

File: packages/core/test/spawn.test.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    );
+    const actor = createActor(
+      createMachine({
+        types: {
+          actors: {} as {
+            src: 'fetchNum';
+            logic: typeof fetchNum;
+          }
+        },
+        entry: spawn('fetchNum', { id: 'child', input: 21 })
```
Comment (Andarist) [inline packages/core/test/spawn.test.ts:37]: I wonder if we could make it easier to spawn N actors at the same time. I was thinking about the API that would allow for a resolver to return many actors:
```ts
spawn(spawn => ({
  id1: spawn('foo', { input: 42 }),
  id2: spawn('bar', { input: 'answer' })
}))
```

Of course, I don't like using `spawn` name for both here. A nice thing about this one is that we could more easily **derive** spawned IDs in the future from types. Although, perhaps we could also do that from the `spawn` action since it returns, well, an action object - and not the actual actor. So we still have a place to keep the inferred ID at the type level. 

So basically... I'm probably just thinking about something that would allow us to spawn multiple actors at the same time without using `pure`

Comment (davidkpiano) [reply packages/core/test/spawn.test.ts:37]: Discussed offline, we'll stick with spawning one at a time (there's `pure` for dynamic spawns)

File: packages/core/src/actions/spawn.ts (hunk 1 above)
Comment (Andarist) [reply packages/core/src/actions/spawn.ts:167]: I managed to deduplicate those

**Review by Andarist — APPROVED** [review]
